api/src/Validator/ContentNode/AssertBelongsToSameRootValidator.php

```diff
@@ -3,34 +3,39 @@
 namespace App\Validator\ContentNode;
 
 use App\Entity\ContentNode;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 use Symfony\Component\Validator\Exception\UnexpectedTypeException;
 use Symfony\Component\Validator\Exception\UnexpectedValueException;
 
-class AssertBelongsToSameOwnerValidator extends ConstraintValidator {
-    public function __construct() {
+class AssertBelongsToSameRootValidator extends ConstraintValidator {
+    public function __construct(public RequestStack $requestStack) {
     }
 
     public function validate($value, Constraint $constraint) {
-        if (!$constraint instanceof AssertBelongsToSameOwner) {
-            throw new UnexpectedTypeException($constraint, AssertBelongsToSameOwner::class);
-        }
-
-        if (null === $value || '' === $value) {
-            return;
+        if (!$constraint instanceof AssertBelongsToSameRoot) {
+            throw new UnexpectedTypeException($constraint, AssertBelongsToSameRoot::class);
         }
 
         $object = $this->context->getObject();
         if (!$object instanceof ContentNode) {
             throw new UnexpectedValueException($object, ContentNode::class);
         }
 
+        $previousObject = $this->requestStack->getCurrentRequest()?->attributes?->get('previous_data');
+        $previousValue = $previousObject?->{$this->context->getPropertyName()};
+
+        // root nodes have parent:null (before and after)
+        if ($previousObject instanceof ContentNode && null === $value && null === $previousValue) {
+            return;
+        }
```

Comment:
Does a root content node have `$this->root === null`? Would things be easier if we had a dedicated `RootContentNode` class which just returns `$this` when asked for `getRoot()`?

Reply:
Root nodes have `$parent===null` and `$root === $this` (same as before).

`$root === null` could also be an option for root nodes. 

Implementing a dedicated `RootContentNode` could be an option, yes.